extensions/iceberg/src/main/java/io/deephaven/iceberg/layout/IcebergFlatLayout.java

```diff
@@ -0,0 +1,80 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.layout;
+
+import io.deephaven.base.FileUtils;
+import io.deephaven.engine.table.impl.locations.TableDataException;
+import io.deephaven.engine.table.impl.locations.impl.TableLocationKeyFinder;
+import io.deephaven.iceberg.location.IcebergTableLocationKey;
+import org.apache.iceberg.*;
+import org.apache.iceberg.io.FileIO;
+import org.jetbrains.annotations.NotNull;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+/**
+ * Iceberg {@link TableLocationKeyFinder location finder} for tables without partitions that will discover data files
+ * from a {@link org.apache.iceberg.Snapshot}
+ */
+public final class IcebergFlatLayout implements TableLocationKeyFinder<IcebergTableLocationKey> {
```

Comment:
The parquet naming convention is `ParquetFlatPartitionedLayout` and `ParquetKeyValuePartitionedLayout`. You can follow the same here for iceberg.